bgpd/bgp_attr.c

```diff
@@ -1628,10 +1628,15 @@ bgp_attr_malformed(struct bgp_attr_parser_args *args, uint8_t subcode,
 		zlog_debug("%s: attributes: %s", __func__, str);
 	}
 
-	/* Only relax error handling for eBGP peers */
-	if (peer->sort != BGP_PEER_EBGP) {
-		bgp_notify_send_with_data(peer->connection,
-					  BGP_NOTIFY_UPDATE_ERR, subcode,
+	/* If the Length of Next Hop Network Address field of the MP_REACH
+	 * attribute is inconsistent with that which was expected, the attribute
+	 * is considered malformed.  Since the next hop precedes the NLRI field
+	 * in the attribute, in this case it will not be possible to reliably
+	 * locate the NLRI; thus, the "session reset" or "AFI/SAFI disable"
+	 * approach MUST be used.
+	 */
+	if (args->type == BGP_ATTR_MP_REACH_NLRI || args->type == BGP_ATTR_MP_UNREACH_NLRI) {
+		bgp_notify_send_with_data(peer->connection, BGP_NOTIFY_UPDATE_ERR, subcode,
```

Comment:
nit: is the alignment correct here - it looks like (in the github ui) the second line of args isn't aligned to the open-parenthesis?